trustmanager/keyfilestore.go

```diff
@@ -43,10 +53,45 @@ func NewKeyFileStore(baseDir string, passphraseRetriever passphrase.Retriever) (
 		return nil, err
 	}
 	cachedKeys := make(map[string]*cachedKey)
+	keyIDMap := make(keyIDMap)
 
-	return &KeyFileStore{SimpleFileStore: *fileStore,
+	keyStore := &KeyFileStore{SimpleFileStore: *fileStore,
 		Retriever:  passphraseRetriever,
-		cachedKeys: cachedKeys}, nil
+		cachedKeys: cachedKeys,
+		keyIDMap:   keyIDMap,
+	}
+
+	// Load this keystore's ID --> gun/role map
+	keyStore.loadKeyIDInfo()
+
+	return keyStore, nil
+}
+
+func generateKeyInfoMap(fullIDToRole map[string]string) map[string]KeyInfo {
+	keyInfoMap := make(map[string]KeyInfo)
+	for keyIDAndGun, role := range fullIDToRole {
+		keyGun := getGunFromFullID(keyIDAndGun)
+		keyID := filepath.Base(keyIDAndGun)
```

Comment:
`listkeys` will handle the old-style filename and clean up the path by the time the code gets here